sdk/scheduler/src/main/java/com/mesosphere/sdk/specification/yaml/YAMLToInternalMappers.java

```diff
@@ -108,7 +111,43 @@ private static void verifyDistinctDiscoveryPrefixes(Collection<RawPod> rawPods)
         }
     }
 
-    private static ReadinessCheckSpec from(RawReadinessCheck rawReadinessCheck) {
+    /**
+     * Verifies that tasks in separate pods don't share advertised port names.
+     * Otherwise the 'endpoints' command will have them jumbled together.
+     */
+    private static void verifyDistinctPortNames(Collection<RawPod> rawPods) {
+        Set<String> portNames = new HashSet<>();
+        Set<String> portNameDuplicates = new TreeSet<>();
+        for (RawPod pod : rawPods) {
+            // Check across both task ports, and resource set ports:
+            for (RawTask task : pod.getTasks().values()) {
+                collectKeys(task.getPorts(), portNames, portNameDuplicates);
+            }
+            if (pod.getResourceSets() != null) {
+                for (RawResourceSet resourceSet : pod.getResourceSets().values()) {
+                    collectKeys(resourceSet.getPorts(), portNames, portNameDuplicates);
+                }
+            }
+        }
+        if (!portNameDuplicates.isEmpty()) {
+            //TODO(nickbp): Only check this when ports are flagged with 'advertise: true' (the current default)
+            throw new IllegalArgumentException(String.format(
+                    "Duplicate port/endpoint names across different tasks: %s", portNameDuplicates));
+        }
+    }
+
+    private static <T> void collectKeys(Map<String, T> map, Set<String> seenNames, Set<String> duplicates) {
```

Comment:
Forgot to generify the variable names when generifying the function. Will do so in the following PR as this one is intended to make the upcoming cut